Commands/help.py

```diff
@@ -12,24 +12,45 @@ def get_available_commands():
     return {"help": execute}
 
 
+def filter_commands_by_permissions(client, command_list):
+    filtered_commands = dict()
+    # check to see if the bot is allowed to run a command
+    if '*' in client.current_bot_permissions['bot-functions']:
+        # all commands allowed
+        logger.debug("All commands permitted. Returning full command list.")
+        filtered_commands = command_list
+
+    else:
+        for command in command_list.keys():
+            if command in client.current_bot_permissions['bot-functions']:
+                #command is allowed
+                logger.debug("Current permissions allow for command '{}'.".format(command))
+                filtered_commands[command] = command_list[command]
+
+    return filtered_commands
+
+
 async def execute(client, message, instruction, **kwargs):
     """Enumerate commands, or explain the functionality of specific command."""
 
-    commands = kwargs['commands']
+    #commands = kwargs['commands']
+    commands = filter_commands_by_permissions(client, kwargs['commands'])
+    logger.info(commands.keys())
 
     if len(instruction) > 2:
         logger.warning("Unknown command '%s'!", instruction[1:])
 
-        await client.send_message(message.author, "Warning: Too many arguments.\n*help*: {}".format(commands['help'].__doc__))
+        await client.send_message(message.author, "Warning: Too many arguments. Try again with the format `b! help [command].`")
```

Comment:
Technically, the invocation is variable and stored in the client (i think).

Reply:
```suggestion
        await client.send_message(message.author, "Warning: Too many arguments. Try again with the format `{} help [command].`".format(client.invocation))
```